Tests/ParseSwiftTests/ParseUserTests.swift

```diff
@@ -1083,59 +1083,75 @@ class ParseUserTests: XCTestCase { // swiftlint:disable:this type_body_length
     func testLogout() {
         testLogin()
         MockURLProtocol.removeAll()
+        DispatchQueue.main.async {
+            let oldInstallationId = BaseParseInstallation.current?.installationId
+            let logoutResponse = NoBody()
 
-        let logoutResponse = NoBody()
-
-        MockURLProtocol.mockRequests { _ in
+            MockURLProtocol.mockRequests { _ in
+                do {
+                    let encoded = try ParseCoding.jsonEncoder().encode(logoutResponse)
+                    return MockURLResponse(data: encoded, statusCode: 200, delay: 0.0)
+                } catch {
+                    return nil
+                }
+            }
             do {
-                let encoded = try ParseCoding.jsonEncoder().encode(logoutResponse)
-                return MockURLResponse(data: encoded, statusCode: 200, delay: 0.0)
+                try User.logout()
+                if let userFromKeychain = BaseParseUser.current {
+                    XCTFail("\(userFromKeychain) wasn't deleted from Keychain during logout")
+                }
+                DispatchQueue.main.async {
+                    if let installationFromKeychain = BaseParseInstallation.current {
+                        if installationFromKeychain.installationId == oldInstallationId {
```

Comment:
What do you think about extending the test case here and also checking if `installationFromKeychain.currentInstallation != nil`?